components/elm/src/biogeochem/EcosystemDynMod.F90

```diff
@@ -233,35 +238,43 @@ subroutine EcosystemDynLeaching(bounds, num_soilc, filter_soilc, &
           call c14_veg_cf%Summary(bounds, num_soilp, filter_soilp, num_soilc, filter_soilc, 'c14', c14_col_cf)
           call c14_col_cf%Summary(bounds, num_soilc, filter_soilc, 'c14')
        end if
-
        call veg_cs%Summary(bounds, num_soilc, filter_soilc, num_soilp, filter_soilp, col_cs)
-       call col_cs%Summary(bounds, num_soilc, filter_soilc)
        if ( use_c13 ) then
           call c13_veg_cs%Summary(bounds, num_soilc, filter_soilc, num_soilp, filter_soilp, c13_col_cs)
           call c13_col_cs%Summary(bounds, num_soilc, filter_soilc)
        end if
        if ( use_c14 ) then
           call c14_veg_cs%Summary(bounds, num_soilc, filter_soilc, num_soilp, filter_soilp, c14_col_cs)
           call c14_col_cs%Summary(bounds, num_soilc, filter_soilc)
-
        end if
-
        call veg_nf%Summary(bounds, num_soilc, filter_soilc, num_soilp, filter_soilp, col_nf)
-       call col_nf%Summary(bounds, num_soilc, filter_soilc)
-
        call veg_ns%Summary(bounds, num_soilc, filter_soilc, num_soilp, filter_soilp, col_ns)
-       call col_ns%Summary(bounds, num_soilc, filter_soilc)
-
        call veg_pf%Summary(bounds, num_soilc, filter_soilc, num_soilp, filter_soilp, col_pf)
-       call col_pf%Summary(bounds, num_soilc, filter_soilc)
-       
        call veg_ps%Summary(bounds, num_soilc, filter_soilc, num_soilp, filter_soilp, col_ps)
-       call col_ps%Summary(bounds, num_soilc, filter_soilc)
-
-       call t_stopf('CNPsum')
-
-    end if !end of if not use_fates block
-
+    else
+       ! In this scenario, we simply zero all of the 
+       ! column level variables that would had been upscaled
+       ! in the veg summary with p2c
+       call col_cs%ZeroUpscaled(bounds,num_soilc, filter_soilc)
+       call col_ns%ZeroUpscaled(bounds,num_soilc, filter_soilc)
+       call col_ps%ZeroUpscaled(bounds,num_soilc, filter_soilc)
+       call col_cf%ZeroUpscaled(bounds,num_soilc, filter_soilc)
+       call col_nf%ZeroUpscaled(bounds,num_soilc, filter_soilc)
+       call col_pf%ZeroUpscaled(bounds,num_soilc, filter_soilc)
```

Comment:
How about renaming `%ZeroUpscaled` to `%ZeroForFATES`?

Reply:
ok, works for me, will do